android/build.gradle

```diff
@@ -9,24 +9,21 @@ buildscript {
         }
     }
     dependencies {
-        classpath 'com.android.tools.build:gradle:4.2.0'
+        classpath 'com.android.tools.build:gradle:8.2.2'
+        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:1.9.21"
 
         // NOTE: Do not place your application dependencies here; they belong
         // in the individual module build.gradle files
-        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.4.32'
     }
 }
 
 allprojects {
     repositories {
         google()
         mavenCentral()
-        flatDir {
-            dirs 'cronet' // use .aar in cronet folder as repo to avoid new gradle error
-        }
```

Comment:
This was included to support building with a local .aar file during development, is that supported in some other way in the current version of gradle?

Reply:
This is deprecated and replaced by

https://github.com/greatfire/envoy/pull/83/files#diff-4f5f18e3075fcd475e12e10a244e29b76785bed75bc03adf41a1067320f3ab35R23-R28

and 

```gradle
implementation (files("cronet/cronet-debug.aar"))
```

See
https://stackoverflow.com/questions/68215302/using-flatdirs-should-be-avoided-because-it-doesnt-support-any-meta-data-format

Reply:
I've used that elsewhere but it had to be changed here for some reason.  I'll try to do a build today and see if I can recall the issue.

Reply:
So, what you want to say is we gotta keep the deprecated `flatDir` for now?

Reply:
As long as I can build an Envoy with the Cronet I just built without having to upload it to Maven, that's all I care about :) It seems like not a totally weird use case, I just don't know the right way to do it. Maybe there isn't one anymore because it's a "broken AAR"? Are we supposed to upload dev builds to Maven? Jam them in to a single AAR?

This SO answer looks promising: https://stackoverflow.com/a/70074787/1376296